File: ios/ScreenshotToggler.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ScreenshotToggler, NSObject)
RCT_EXTERN_METHOD(setScreenshotEnabled:(BOOL)enabled
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/ScreenshotToggler.swift
@objc(ScreenshotToggler)
class ScreenshotToggler: NSObject {
  @objc
  func setScreenshotEnabled(_ enabled: Bool, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    DispatchQueue.main.async {
      if let window = UIApplication.shared.keyWindow {
        if enabled {
          window.layer.contents = nil
          resolve("Success")
        } else {
          window.layer.contents = UIImage().cgImage
          resolve("Success")
        }
      } else {
        reject("NO_WINDOW", "Could not find key window", nil)
      }
    }
  }
}
